src/mbgl/style/expression/format_expression.cpp

```diff
@@ -34,68 +36,69 @@ using namespace mbgl::style::conversion;
 
 ParseResult FormatExpression::parse(const Convertible& value, ParsingContext& ctx) {
     std::size_t argsLength = arrayLength(value);
-    if (argsLength < 3) {
```

Comment:
This will essentially break the backwards compatibility with the text format section correct?

Reply:
It could be done maybe if the next is not an object, implicitly do a text section, no?

Reply:
Backward compatibility is preserved for `non-error` cases. Do you think we should maintain backward compatibility for error path? E.g., someone is relying symbol layer to be not-renderable by intentionally providing malformed expression?

Reply:
Thanks for explaining. The current implementation seems like a good compromise.